geomstats/geometry/discrete_curves.py

```diff
@@ -15,6 +15,7 @@
 from geomstats.geometry.fiber_bundle import AlignerAlgorithm, FiberBundle
 from geomstats.geometry.hypersphere import Hypersphere
 from geomstats.geometry.landmarks import Landmarks
+from geomstats.geometry.manifold import register_quotient_structure
```

Comment:
NIT: I wonder if register_quotient_structure should be "hidden" in manifold.

More generally, I feel that manifold.py and base.py might be hidden important code. base.py has LevelSet, ImmersedSet, etc if I remember correctly.

I would recommend having: manifold.py, level_set.py, immersed_set.py, quotient.py, etc visible.

Not for this PR though, but to be discussed?

Reply:
What do you mean by 

> register_quotient_structure should be "hidden" in manifold"

I kept it public because the method can potentially be used by users without having to import `Manifold` (a bit like the similar method @johnharveymath developed in #1964).


I think the other points are relevant. We should probably split `base` into multiple files in a future PR. We may need to check if there will be no circular imports (though I believe not).

Reply:
Sorry I wasn't clear. I didn't mean public/private, but rather separated in a different file. I feel that manifold.py and base.py contain too many important classes, and we should probably split both manifold.py and base.py into several files.

Also, we should look into our init.py and import strategies, i.e., starting using `__all__` and other syntaxes like that to ease imports.

Not a priority, some next PR!